src/pkg/term/colorizer.go

```diff
@@ -12,6 +12,54 @@ import (
 	"golang.org/x/term"
 )
 
+// DefaultBufferSize is the default number of recent messages to keep in the circular buffer
+var DefaultBufferSize = 10
+
+type circularBuffer struct {
+	size         int
+	linesWritten int
+	index        int
+	data         []string
+}
+
+func (c *circularBuffer) write(msg string) {
+	c.linesWritten++
+	if len(c.data) < c.size {
+		c.data = append(c.data, msg)
+	} else {
+		c.data[c.index] = msg
+	}
```

Comment:
I wouldn't bother maintaining two branches here. This is so subtle. We know how big we want the buffer to be, so let's just allocate the whole thing, then use `c.data[c.index] = msg` all the time.